### ISSUE TITLE
GitHub Actions rate limiting, GITHUB_TOKEN, thoughts?

### ISSUE THREAD

**Issue by guzba** [jiro4989/setup-nim-action#296]
Hello and thanks for the work you've done with this repo, it has worked very well for a long time now.

One thing I've started seeing far more often is rate limit issues. I'm wondering if there are any changes made to this repo that could help mitigate them. I did some searching and I didn't see any other issues related to rate limiting so I have decided to create one.

First, this is not an error from this repo. I am just wondering if a change to this repo could help reduce the frequency or if this is actually caused by something else.

An example of what I see:

```
Run jiro4989/setup-nim-action@v1
  with:
    nim-version: 1.4.x
    no-color: false
    yes: false
Fetch a latest versions with 1.4.x
Error: 403 - {"message":"API rate limit exceeded for 13.83.3.161. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)","documentation_url":"https://docs.github.com/rest/overview/resources-in-the-rest-api#rate-limiting"}
```

The only request I see to GitHub is https://github.com/jiro4989/setup-nim-action/blob/29377273a9bceffbaa9099d21339fe6fc5f1444e/src/util.ts#L18

Would including a header of Authorization: Bearer GITHUB_TOKEN there help possibly? Or maybe you know more about this issue and would be able to share more about what is happening.

**Comment by jiro4989** [comment]
Thank you. I was also aware of the rate limiting error.
I was wondering how to fix it. For the time being, I will try to solve the error with a bearer token.

**Comment by jiro4989** [comment]
@guzba I released [v1.4.0](https://github.com/jiro4989/setup-nim-action/releases/tag/v1.4.0).
Please set `repo-token` parameter and retry setup-nim-action.

**Comment by guzba** [comment]
I was able to get this added here: https://github.com/guzba/zippy/commit/123cd593f513d584526f119e7ccb65a7baefa8fa

Was easy to do thanks to your updated readme basic usage example.

The run for that push worked the first try which was starting to get rare so I think this helped!

Thanks for the quick support of the token.
